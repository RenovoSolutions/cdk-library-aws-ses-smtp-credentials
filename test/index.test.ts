import {
  Stack,
  App,
} from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {
  SesSmtpCredentials,
} from '../src/index';

test('Snapshot', () => {
  const app = new App();
  const stack = new Stack(app, 'TestStack');

  new SesSmtpCredentials(stack, 'SesSmtpCredentials', {
    iamUserName: 'exampleUser',
  });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});