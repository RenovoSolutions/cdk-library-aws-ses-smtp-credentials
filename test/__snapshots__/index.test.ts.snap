// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "SesSmtpCredentialsOnEventHandlerBEA768EC": Object {
      "DependsOn": Array [
        "SesSmtpCredentialsRole6AB00532",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "7224866b0a593c509766adc7e7d3386a8672fd9077269c679594700a955d44d1.zip",
        },
        "Handler": "index.on_event",
        "Role": Object {
          "Fn::GetAtt": Array [
            "SesSmtpCredentialsRole6AB00532",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "SesSmtpCredentialsProviderframeworkonEvent771E2E12": Object {
      "DependsOn": Array [
        "SesSmtpCredentialsProviderframeworkonEventServiceRoleDefaultPolicy7987CCED",
        "SesSmtpCredentialsProviderframeworkonEventServiceRoleCEC286EB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "2157f4ab8972014e220d70707296b292b3b7301f163f7cd641ccda0ee663530f.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (TestStack/SesSmtpCredentials/Provider)",
        "Environment": Object {
          "Variables": Object {
            "USER_ON_EVENT_FUNCTION_ARN": Object {
              "Fn::GetAtt": Array [
                "SesSmtpCredentialsOnEventHandlerBEA768EC",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": Object {
          "Fn::GetAtt": Array [
            "SesSmtpCredentialsProviderframeworkonEventServiceRoleCEC286EB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "SesSmtpCredentialsProviderframeworkonEventServiceRoleCEC286EB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SesSmtpCredentialsProviderframeworkonEventServiceRoleDefaultPolicy7987CCED": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "SesSmtpCredentialsOnEventHandlerBEA768EC",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "SesSmtpCredentialsOnEventHandlerBEA768EC",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "SesSmtpCredentialsProviderframeworkonEventServiceRoleDefaultPolicy7987CCED",
        "Roles": Array [
          Object {
            "Ref": "SesSmtpCredentialsProviderframeworkonEventServiceRoleCEC286EB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SesSmtpCredentialsRole6AB00532": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Ref": "SesSmtpCredentialsSecretsManagerPolicyC597B13A",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SesSmtpCredentialsSecretArn7500943E": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "SesSmtpCredentialsUserE9E5FC46",
      ],
      "Properties": Object {
        "KmsKeyId": "aws/secretsmanager",
        "Override": true,
        "Region": Object {
          "Ref": "AWS::Region",
        },
        "Restore": true,
        "SecretName": "TestStackSesSmtpCredentials1AD8F67EexampleUser",
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "SesSmtpCredentialsProviderframeworkonEvent771E2E12",
            "Arn",
          ],
        },
        "UserName": "exampleUser",
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "SesSmtpCredentialsSecretResourcePolicyFA18D550": Object {
      "Properties": Object {
        "ResourcePolicy": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Condition": Object {
                "ArnLike": Object {
                  "aws:SourceArn": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:aws:rds:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":og:teststack*",
                      ],
                    ],
                  },
                },
                "StringEquals": Object {
                  "aws:sourceAccount": Array [
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                  ],
                },
              },
              "Effect": "Allow",
              "Principal": Object {
                "Service": "rds.amazonaws.com",
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "SecretId": Object {
          "Fn::GetAtt": Array [
            "SesSmtpCredentialsSecretArn7500943E",
            "SecretArn",
          ],
        },
      },
      "Type": "AWS::SecretsManager::ResourcePolicy",
    },
    "SesSmtpCredentialsSecretsManagerPolicyC597B13A": Object {
      "Properties": Object {
        "Description": "",
        "Path": "/",
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "secretsmanager:PutSecretValue",
                "secretsmanager:CreateSecret",
                "secretsmanager:DeleteSecret",
                "secretsmanager:UpdateSecret",
                "secretsmanager:TagResource",
                "secretsmanager:RestoreSecret",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:secretsmanager:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":secret:TestStackSesSmtpCredentials1AD8F67EexampleUser-*",
                  ],
                ],
              },
              "Sid": "SecretsManagerPolicy",
            },
            Object {
              "Action": Array [
                "iam:CreateAccessKey",
                "iam:DeleteAccessKey",
                "iam:ListAccessKeys",
              ],
              "Condition": Object {
                "StringEquals": Object {
                  "iam:ResourceTag/CfnStackIdForSesCredLibrary": Object {
                    "Ref": "AWS::StackId",
                  },
                },
              },
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "IamAllowKeyManagementPolicy",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
    "SesSmtpCredentialsSesSendPolicyDCC9F0D8": Object {
      "Properties": Object {
        "Description": "",
        "Path": "/",
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "ses:SendRawEmail",
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "SesAllowSendPolicy",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
    "SesSmtpCredentialsUserE9E5FC46": Object {
      "Properties": Object {
        "ManagedPolicyArns": Array [
          Object {
            "Ref": "SesSmtpCredentialsSesSendPolicyDCC9F0D8",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "CfnStackIdForSesCredLibrary",
            "Value": Object {
              "Ref": "AWS::StackId",
            },
          },
        ],
        "UserName": "exampleUser",
      },
      "Type": "AWS::IAM::User",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
